modules/thermo-cycler/production/firmware_uploader.py

```diff
@@ -1,20 +1,31 @@
-# This script searches for a thermocycler port and uploads the pre-built binary
-# from /build/tmp to the module
-# Usage: `python firmware_uploader.py` (No args)
+# This script searches for a thermocycler port and uploads the pre-built .bin
+# from `/production/firmware`, or, a .bin/.uf2 file provided, to the module
+#
+# Usage: `python firmware_uploader.py` or
+#        `python firmware_uploader.py -F my_firmware_file.bin`
 # Requires:
 #   - New uf2 bootloader for thermocycler
-#   - Firmware binary in /build/tmp (This is where `make build` or Arduino->compile
-#     would store the build binary)
+#   - Firmware binary in `production/firmware/`
+#     or path to binary/uf2 file added as argument `-F`
 
+import os
 import uf2conv
 import serial
 import time
 from serial.tools.list_ports import comports
+from argparse import ArgumentParser
 
-FIRMWARE_FILE_PATH = "../../../build/tmp/thermo-cycler-arduino.ino.bin"
+THIS_DIR = os.path.dirname(os.path.realpath(__file__))
+DEFUALT_FW_FILE_PATH = os.path.join(THIS_DIR, "firmware", "thermo-cycler-arduino.ino.bin")
 OPENTRONS_VID = 1240
 MAX_SERIAL_LEN = 16
 
+def build_arg_parser():
+    arg_parser = ArgumentParser(
+        description="Thermocycler firmware uploader")
+    arg_parser.add_argument("-F", "--fw_file", required=False)
```

Comment:
if you put the default above into here as `default=DEFAULT_FW_FILE_PATH` it will display when you do `--help` but this works fine too

Reply:
oh! cool. I'll make that change